### ISSUE TITLE
The cross function has a bug

### ISSUE THREAD

**Issue by casper-hansen** [twopirllc/pandas-ta#560]
When both inputs are zero or null, it will always say there is a cross below and never a cross above on two series of data.

```python
ta.cross(df.conversion_line, df.base_line, above=True)
ta.cross(df.conversion_line, df.base_line, above=False)
```

![image](https://user-images.githubusercontent.com/27340033/177002051-d3da04b6-342d-430e-854d-dd57f452b761.png)

**Comment by casperbh96** [comment]
I implemented a quick fix for when values are null. The real solution to this bug should also include if both are zero.

```python
def cross(series_a, series_b, asint=True, above=True):
    # Calculate Result
    current = series_a > series_b  # current is above
    previous = series_a.shift(1) < series_b.shift(1)  # previous is below
    series_a_null = pd.isnull(series_a)
    series_b_null = pd.isnull(series_b)

    # above if both are true, below if both are false
    crossed = current & previous & ~series_a_null & ~series_b_null if above else ~current & ~previous & ~series_a_null & ~series_b_null 

    if asint:
        crossed = crossed.astype(int)

    return crossed
```

**Comment by twopirllc** [comment]
Hello @casperbh96,

Right. I am aware of this and Issue #500. Does your fix also help resolve #500? Also I am trying to improve performance for future releases, such as the _development_ branch, with _numpy/numba_ versions. So if you could make a _numpy_ version, that would be appreciative. 😎 

Kind Regards,
KJ

**Comment by casperbh96** [comment]
> Hello @casperbh96,
> 
> Right. I am aware of this and Issue #500. Does your fix also help resolve #500? Also I am trying to improve performance for future releases, such as the _development_ branch, with _numpy/numba_ versions. So if you could make a _numpy_ version, that would be appreciative. 😎
> 
> Kind Regards, KJ

No, it does not solve the issue as none of your code takes null values into account. Numpy version is not hard to implement if you have the time.

**Comment by twopirllc** [comment]
Thanks

**Comment by twopirllc** [comment]
Please see #901
